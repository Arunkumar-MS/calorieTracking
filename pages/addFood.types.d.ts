export interface FoodEntry {
    _id?: string;
    servingQty: number;
    servingUnit: string;
    servingWeightGrams: string;
    url: string;
    calories: number;
    name: string;
    userId: string;
    addedDate: number;
}