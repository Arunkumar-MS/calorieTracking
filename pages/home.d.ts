type Role = 'admin' | 'user' | null;

export interface User {
    name: string;
    userId: string;
    role: Role;
}

export interface Props {
    user: User;
}