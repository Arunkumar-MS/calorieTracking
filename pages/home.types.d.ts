type Role = 'admin' | 'user' | null;

export interface User {
    name: string;
    userId: string;
    role: Role;
    calorieLimit: number;
}

export interface Props {
    user: User;
}