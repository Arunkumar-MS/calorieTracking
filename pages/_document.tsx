import Document, { Html, Head, Main, NextScript, DocumentContext } from 'next/document'

class MyDocument extends Document {
  static async getInitialProps(ctx: DocumentContext) {
    const initialProps = await Document.getInitialProps(ctx)
    return { ...initialProps }
  }

  render() {
    return (
      <Html className='h-full bg-gray-100' lang="en-us">
        <title>
          Caloriese tracker
        </title>
        <meta name="description" content="This is caloriese tracking app. which helps you to keep fit by setting caloriese limte per day."/>
        <Head />
        <body className='h-full'>
          <Main />
          <NextScript />
        </body>
      </Html>
    )
  }
}

export default MyDocument