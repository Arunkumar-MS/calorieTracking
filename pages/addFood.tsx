import type { NextPage } from 'next';
import React from 'react';
import Layout from '@Layout/index';
import { useForm } from "react-hook-form";
import { useRouter } from 'next/router';
import {
    useAppDispatch,
} from '@Store/hooks';
import { updateUser } from '@Reducers/userSlice/userSlice';
import { Props as InitailProps } from '@Pages/home.types';
import authAccess from '@Service/serverAuth';
import axios from '@Service/core/axios';
import { FoodEntry } from '@Pages/addFood.types';
import { updateEntry } from '@Reducers/foodDetailsSlice/foodDetailsSlice';

type Props = NextPage & InitailProps;


const saveFood = (data: FoodEntry) => {
    return axios.post('/tracker/addFood', { ...data });
}


const AddFood = (props: Props) => {
    const {
        register,
        handleSubmit,
        formState,
        watch,
        formState: { errors }
    } = useForm();

    const dispatch = useAppDispatch();

    const { query } = useRouter();

    React.useEffect(() => {
        dispatch(updateUser(props.user))
    }, []);

    const onSubmit = (data: any) => {
        saveFood(data).then((list) => dispatch(updateEntry(list.data)));
    };

    const getFormRegisterPropsForString = (name: string) => {
        return {
            ...register(name, {
                required: true,
                maxLength: 150,
                pattern: /^[A-Za-z]+$/i
            })
        }
    }

    const getFormRegisterPropsForURL = (name: string) => {
        return {
            ...register(name, {
                required: true,
                maxLength: 2045,
                pattern: /(http(s)?:\/\/.)?(www\.)?[-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_\+.~#?&//=]*)/g
            })
        }
    }

    const getFormRegisterPropsForNumber = (name: string) => {
        return {
            ...register(name, {
                required: true,
                maxLength: 150,
                pattern: /^[0-9]+$/i
            })
        }
    }


    const renderRow = (name: string, registerProps: any, error: any, exp?: string, defultValue?: any, className?: string) => {
        return (
            <div className={`sm:grid sm:grid-cols-3 sm:gap-4 sm:items-start sm:border-t sm:border-gray-200 pt-2 pb-2 sm:pt-5  sm:pb-5 ${className}`}>
                <label htmlFor="last-name" className="block text-sm font-medium text-gray-700 sm:mt-px sm:pt-2">
                    <span>{name}</span>
                    <span className='text-xs'>{exp}</span>
                </label>
                <div>
                    <div className="mt-1 sm:mt-0 sm:col-span-2">

                        <input
                            {...registerProps}
                            type="text"
                            id="last-name"
                            defaultValue={defultValue}
                            className="max-w-lg block w-full shadow-sm focus:ring-indigo-500 focus:border-indigo-500 sm:max-w-xs sm:text-sm border-gray-300 rounded-md"
                        />
                    </div>
                    {error && <span className="flex items-center font-medium tracking-wide text-red-500 text-xs mt-1 ml-1">
                        {`Please enter valid ${name}!`}
                    </span>}
                </div>
            </div>
        )

    }


    return (
        <Layout>
            <form onSubmit={handleSubmit(onSubmit)}>
                {renderRow('Food name', getFormRegisterPropsForString('name'), errors['name'], '(Pizza)', query.foodname)}
                {renderRow('Consume qty  ', getFormRegisterPropsForNumber('consumeQty'), errors['consumeQty'], '(1)')}
                {renderRow('Consumed Calories', getFormRegisterPropsForNumber('consumedCalories'), errors['consumedCalories'], '(120)')}

                {renderRow('Serving unit ', getFormRegisterPropsForString('servingUnit'), errors['servingUnit'], '(Slice)')}
                {renderRow('Consumed weight grams ', getFormRegisterPropsForNumber('consumedWeightGrams'), errors['consumedWeightGrams'], '(107)')}
                {renderRow('Image Url', getFormRegisterPropsForURL('imageUrl'), errors['imageUrl'])}
                <div className=' w-full flex justify-center'>
                    <button type="submit" className="w-full md:w-5/12 mt-2 bg-blue-500 rounded-lg font-bold text-white text-center px-4 py-3 transition duration-300 ease-in-out hover:bg-blue-600">
                        Add
                    </button>
                </div>

            </form>
        </Layout>

    )
}

export async function getServerSideProps(ctx: any) {
    const props = await authAccess(ctx);
    return props;
}

export default AddFood;
