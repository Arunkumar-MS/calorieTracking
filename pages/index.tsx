import * as React from 'react';
import {
  useAppDispatch,
} from '@Store/hooks';
import UserFlow from '@Component/userFlow/userFlow';
import { updateEntry } from '@Reducers/foodDetailsSlice/foodDetailsSlice';
import Spinner from '@Component/spinner';
import useSWR from 'swr';
import withAuth from 'src/authHoc';
import { SWR_OPTIONS } from 'src/constant/swr';
import fetcher from '@Service/core/fetcher';


export const Home = () => {

  const dispatch = useAppDispatch();
  const { data, error } = useSWR('/tracker/getfoodEntry', fetcher, { ...SWR_OPTIONS });
  React.useEffect(() => {
    if(data) {
      dispatch(updateEntry(data));
    }
  }, [data]);

  return (
    <>
      {!data && !error && <Spinner className='mt-20'/>}
      {data && <UserFlow />}
      {error && <div className='text-center text-gray-500'> Something went worng please try later! </div>}
    </>

  )
}


export default withAuth(Home, ['admin','user']);
