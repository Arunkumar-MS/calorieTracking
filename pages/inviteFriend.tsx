import { useForm } from "react-hook-form";
import React from "react";
import { useRouter } from 'next/router'
import withAuth from "src/authHoc";
import { AddUser, createUser } from "@Service/user";
import getUnixTime from "date-fns/getUnixTime";
import Spinner from "@Component/spinner";
import { DEFAULT_CALORIES_LIMIT } from "src/constant/config";


const InviteFriendComponent = () => {
    const router = useRouter()
    const [isLoading, setLoading] = React.useState(false);
    const [error, setError] = React.useState('');
    const [user, setUser] = React.useState<Partial<AddUser | null>>(null);
    const {
        register,
        handleSubmit,
        formState: { errors }
    } = useForm();
    const onSubmit = (data: Partial<AddUser>) => {
        setLoading(true);
        const req = {
            ...data,
            createdDate: getUnixTime(new Date()),
            calorieLimit: DEFAULT_CALORIES_LIMIT,
            role: 'user',
        } as AddUser;

        createUser(req)
            .then((res) => {
                if (res.data.status && res.data.status === 406) {
                    setError(res.data.message);
                } else {
                    setUser({ name: req.name, token: res.data.token });
                }
                setLoading(false);
            })
            .catch(() => {
                setLoading(false);
                setError('Something went wrong please try later!');
            });
    };


    const getFormRegisterPropsForString = (name: string) => {
        return {
            ...register(name, {
                required: true,
                maxLength: 150,
                pattern: /^[A-Za-z0-9 _]+$/i
            })
        }
    }

    const getFormRegisterPropsForEmail = (name: string) => {
        return {
            ...register(name, {
                required: true,
                maxLength: 2045,
                pattern: /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,5})+$/i
            })
        }
    }

    const copy = () => {
        navigator.clipboard.writeText(user?.token || '');
    }


    return (
        <div className="py-16 sm:py-24 flex justify-center">
            <div className="relative sm:py-16 md:w-2/4" >
                <div className="mx-auto max-w-md px-4 sm:max-w-3xl sm:px-6 lg:max-w-7xl lg:px-8">
                    <div className="relative rounded-2xl px-6 py-10 bg-indigo-600 overflow-hidden shadow-xl sm:px-12 sm:py-20">
                        <div aria-hidden="true" className="absolute inset-0 -mt-72 sm:-mt-32 md:mt-0">
                            <svg className="absolute inset-0 h-full w-full" preserveAspectRatio="xMidYMid slice" xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 1463 360">
                                <path className="text-indigo-500 text-opacity-40" fill="currentColor" d="M-82.673 72l1761.849 472.086-134.327 501.315-1761.85-472.086z" />
                                <path className="text-indigo-700 text-opacity-40" fill="currentColor" d="M-217.088 544.086L1544.761 72l134.327 501.316-1761.849 472.086z" />
                            </svg>
                        </div>
                        {!user && (<div className="relative">
                            <div className="text-xl  md:text-2xl text-center font-semibold   antialiased text-white">
                                Invite youre friend
                            </div>
                            <form onSubmit={handleSubmit(onSubmit)}>
                                <div className="flex flex-col">
                                    <div className="">
                                        <input {...getFormRegisterPropsForString("name")} type="text" className="mt-10 block w-full border border-transparent rounded-md px-5 py-3 text-base text-gray-900 placeholder-gray-500 shadow-sm focus:outline-none focus:border-transparent focus:ring-2 focus:ring-white focus:ring-offset-2 focus:ring-offset-indigo-600" placeholder="Enter Name" />
                                        {errors['name'] && <span className="flex items-center font-medium tracking-wide text-red-400 text-xs mt-1 ml-1">
                                            {`Please enter valid name!`}
                                        </span>}
                                        <input  {...getFormRegisterPropsForEmail("emailId")} type="email" className="mt-5 mb-5 block w-full border border-transparent rounded-md px-5 py-3 text-base text-gray-900 placeholder-gray-500 shadow-sm focus:outline-none focus:border-transparent focus:ring-2 focus:ring-white focus:ring-offset-2 focus:ring-offset-indigo-600" placeholder="Enter email id" />
                                        {errors['emailId'] && <span className="flex items-center font-medium tracking-wide text-red-400 text-xs mt-1 ml-1 mb-2">
                                            {`Please enter valid email id!`}
                                        </span>}
                                    </div>
                                    <div className="flex justify-center">
                                        <button disabled={isLoading} type="button" onClick={() => router.back()} className="block w-full mr-2 md:mr-8 md:w-80 rounded-md border border-transparent py-1 px-5 md:py-3 bg-indigo-500 text-base font-medium text-white shadow hover:bg-indigo-400 focus:outline-none focus:ring-2 focus:ring-white focus:ring-offset-2 focus:ring-offset-indigo-600 sm:px-10">Go back</button>
                                        <button disabled={isLoading} type="submit" className="block w-full ml-2 md:w-80 rounded-md border border-transparent px-5 py-3 bg-indigo-500 text-base font-medium text-white shadow hover:bg-indigo-400 focus:outline-none focus:ring-2 focus:ring-white focus:ring-offset-2 focus:ring-offset-indigo-600 sm:px-10">Invite</button>
                                    </div>
                                    {isLoading && <Spinner />}
                                </div>
                            </form>
                            {error &&
                                <div className=" text-center mt-5">
                                    <div className="text-gray-300  font-medium">{error}</div>
                                </div>
                            }
                        </div>)
                        }
                        {user && (
                            <div className="relative">
                                <div className="flex flex-col text-center text-xl  md:text-2xl font-semibold   antialiased text-white">
                                    <div>
                                        {`Share below toekn with youre friend ${user?.name}`}
                                    </div>
                                    <p className="mt-5 mb-5 italic text-gray-100 break-all	text-sm	">
                                        {user?.token}
                                    </p>
                                </div>
                                <div className="flex justify-center">
                                    <button type="button" onClick={() => router.back()} className="block w-full mr-2 md:mr-8 md:w-80 rounded-md border border-transparent py-1 px-5 md:py-3 bg-indigo-500 text-base font-medium text-white shadow hover:bg-indigo-400 focus:outline-none focus:ring-2 focus:ring-white focus:ring-offset-2 focus:ring-offset-indigo-600 sm:px-10">Cancel</button>
                                    <button type="button" onClick={copy} className="block w-full ml-2 md:w-80 rounded-md border border-transparent px-5 py-3 bg-indigo-800 text-base font-medium text-white shadow hover:bg-indigo-400 focus:outline-none focus:ring-2 focus:ring-white focus:ring-offset-2 focus:ring-offset-indigo-600 sm:px-10">Copy to clipboard</button>
                                </div>
                            </div>
                        )}
                    </div>
                </div>
            </div>
        </div>
    )
}


const InviteFriend = withAuth(InviteFriendComponent, ['admin', 'user']);
export default InviteFriend;
