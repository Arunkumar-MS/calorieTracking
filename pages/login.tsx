import axios from "@Service/core/axios";
import React from "react";
import cookies from "js-cookie";
import { useRouter } from 'next/router'


export const Login = () => {
  const router = useRouter()
  const [token, setToken] = React.useState('');
  const [error, setError] = React.useState('');

  const onLoginHandler = () => {
    if(!token) {
      setError('Input field is required. Please enter toekn!');
      return;
    }
    setError('')
    axios.post('/auth/login', { token }).then((d) => {
      cookies.set('token', token);
      router.push('/');
    }).catch((e) => {
      setError('Somthing went wrong. Please try again!')
    });
  }

  const onChangeHandler = (e: React.ChangeEvent<HTMLInputElement>) => {
    setToken(e.target.value);
  }

  return (
    <div className="py-16 sm:py-24" data-test-id="login-page">
      <div className="relative sm:py-16">
        <div className="mx-auto max-w-md px-4 sm:max-w-3xl sm:px-6 lg:max-w-7xl lg:px-8">
          <div className="relative rounded-2xl px-6 py-10 bg-indigo-600 overflow-hidden shadow-xl sm:px-12 sm:py-20">
            <div aria-hidden="true" className="absolute inset-0 -mt-72 sm:-mt-32 md:mt-0">
              <svg className="absolute inset-0 h-full w-full" preserveAspectRatio="xMidYMid slice" xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 1463 360">
                <path className="text-indigo-500 text-opacity-40" fill="currentColor" d="M-82.673 72l1761.849 472.086-134.327 501.315-1761.85-472.086z" />
                <path className="text-indigo-700 text-opacity-40" fill="currentColor" d="M-217.088 544.086L1544.761 72l134.327 501.316-1761.849 472.086z" />
              </svg>
            </div>
            <div className="relative">

              <div className="sm:text-center">
                <h2 className="text-3xl font-extrabold text-white tracking-tight sm:text-4xl" data-test-id="login-page-title">
                  Enter token below for login.
                </h2>
              </div>


              <div className="mt-12 sm:mx-auto sm:max-w-lg sm:flex">

                <div className="min-w-0 flex-1">
                  <label htmlFor="cta-token" className="sr-only">Enter Token</label>
                  <input data-test-id="login-page-token-input" id="cta-token" type="email" className="block w-full border border-transparent rounded-md px-5 py-3 text-base text-gray-900 placeholder-gray-500 shadow-sm focus:outline-none focus:border-transparent focus:ring-2 focus:ring-white focus:ring-offset-2 focus:ring-offset-indigo-600" placeholder="Enter your token" onChange={onChangeHandler} />
                </div>
                <div className="mt-4 sm:mt-0 sm:ml-3">
                  <button data-test-id="login-page-submit" type="button" onClick={onLoginHandler} className="block w-full rounded-md border border-transparent px-5 py-3 bg-indigo-500 text-base font-medium text-white shadow hover:bg-indigo-400 focus:outline-none focus:ring-2 focus:ring-white focus:ring-offset-2 focus:ring-offset-indigo-600 sm:px-10">Login</button>
                </div>
              </div>
              {error && <div className=" text-center mt-5">
                <div className="text-gray-300	font-medium">{error}</div>
              </div>
              }
            </div>
          </div>
        </div>
      </div>
    </div>
  )
}

export default Login;
