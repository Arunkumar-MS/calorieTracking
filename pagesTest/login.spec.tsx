import { jest } from '@jest/globals'
import { shallowToJson } from "enzyme-to-json";
import * as React from 'react';
import { shallow } from 'enzyme';
import { Login } from '../pages/login';

const mockDispatch = jest.fn();
jest.mock('react-redux', () => ({
    useSelector: jest.fn(),
    useDispatch: () => mockDispatch
}));

describe('Login page', () => {
    it('should render properly', function () {
        jest.clearAllMocks();
        const wrap = shallow(<Login />)
        expect(shallowToJson(wrap)).toMatchSnapshot();
    })
})
