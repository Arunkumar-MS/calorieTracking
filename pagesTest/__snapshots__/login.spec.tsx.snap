// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login page should render properly 1`] = `
<div
  className="py-16 sm:py-24"
  data-test-id="login-page"
>
  <div
    className="relative sm:py-16"
  >
    <div
      className="mx-auto max-w-md px-4 sm:max-w-3xl sm:px-6 lg:max-w-7xl lg:px-8"
    >
      <div
        className="relative rounded-2xl px-6 py-10 bg-indigo-600 overflow-hidden shadow-xl sm:px-12 sm:py-20"
      >
        <div
          aria-hidden="true"
          className="absolute inset-0 -mt-72 sm:-mt-32 md:mt-0"
        >
          <svg
            className="absolute inset-0 h-full w-full"
            fill="none"
            preserveAspectRatio="xMidYMid slice"
            viewBox="0 0 1463 360"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              className="text-indigo-500 text-opacity-40"
              d="M-82.673 72l1761.849 472.086-134.327 501.315-1761.85-472.086z"
              fill="currentColor"
            />
            <path
              className="text-indigo-700 text-opacity-40"
              d="M-217.088 544.086L1544.761 72l134.327 501.316-1761.849 472.086z"
              fill="currentColor"
            />
          </svg>
        </div>
        <div
          className="relative"
        >
          <div
            className="sm:text-center"
          >
            <h2
              className="text-3xl font-extrabold text-white tracking-tight sm:text-4xl"
              data-test-id="login-page-title"
            >
              Enter token below for login.
            </h2>
          </div>
          <div
            className="mt-12 sm:mx-auto sm:max-w-lg sm:flex"
          >
            <div
              className="min-w-0 flex-1"
            >
              <label
                className="sr-only"
                htmlFor="cta-token"
              >
                Enter Token
              </label>
              <input
                className="block w-full border border-transparent rounded-md px-5 py-3 text-base text-gray-900 placeholder-gray-500 shadow-sm focus:outline-none focus:border-transparent focus:ring-2 focus:ring-white focus:ring-offset-2 focus:ring-offset-indigo-600"
                data-test-id="login-page-token-input"
                id="cta-token"
                onChange={[Function]}
                placeholder="Enter your token"
                type="email"
              />
            </div>
            <div
              className="mt-4 sm:mt-0 sm:ml-3"
            >
              <button
                className="block w-full rounded-md border border-transparent px-5 py-3 bg-indigo-500 text-base font-medium text-white shadow hover:bg-indigo-400 focus:outline-none focus:ring-2 focus:ring-white focus:ring-offset-2 focus:ring-offset-indigo-600 sm:px-10"
                data-test-id="login-page-submit"
                onClick={[Function]}
                type="button"
              >
                Login
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
